Dockerfile

```diff
@@ -0,0 +1,42 @@
+FROM ubuntu:trusty
+
+LABEL description="BWS (Decred fork)"
+LABEL version="1.0"
+LABEL maintainer "peter@prioritylane.com"
+
+ENV NODE_VERSION v4.2.1
+ENV TERM linux
+ENV USER bws
+ENV APPDIR /home/$USER/app
+
+# create user
+RUN adduser --disabled-password --gecos ''  $USER
+
+# install build tooling
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -qy build-essential python git curl
+
+# create application directory
+RUN mkdir -p $APPDIR && \
+    chown $USER $APPDIR
+
+# switch user
+USER $USER
+WORKDIR $APPDIR
+ENV HOME /home/$USER
+ENV NVM_DIR $HOME/.nvm
+
+COPY . $APPDIR
+
+# install & configure NodeJS
+RUN curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.1/install.sh | bash && \
```

Comment:
As mentioned in similar PRs, I'd really rather a git checkout followed by running a command rather than piping the output of curl straight to bash.